File: Valid Anagram - Leetcode 242/Valid Anagram - Leetcode 242.cpp
#include <unordered_map>
#include <string>
using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }

        unordered_map<char, int> sMap, tMap;

        for (char c : s) {
            sMap[c]++;
        }

        for (char c : t) {
            tMap[c]++;
        }

        return sMap == tMap;
    }
};

File: Valid Anagram - Leetcode 242/Valid Anagram - Leetcode 242.py
from collections import Counter
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        s_dict = Counter(s)
        t_dict = Counter(t)

        return s_dict == t_dict

# Time complexity: O(n)
# Space complexity: O(n)
